File: 3718 - Smallest Missing Multiple of K/Method 1/solution.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var missingMultiple = function(nums, k) {
    const set = new Set(nums);
    let temp = k;
    while(set.has(temp)) {
        temp += k;
    }
    return temp;
};

File: 3718 - Smallest Missing Multiple of K/Method 1/solution.java
class Solution {
    public int missingMultiple(int[] nums, int k) {
        HashSet<Integer> set = new HashSet<>();

        for (int num : nums) {
            set.add(num);
        }

        int temp = k;

        while (set.contains(temp)) {
            temp += k;
        }

        return temp;
    }
}